KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/LiveStreamSettingsModule.java

```diff
@@ -0,0 +1,340 @@
+
+		private IApplicationInstance appInstance = null;
+
+		public LiveStreamPacketizerListener(IApplicationInstance appInstance) {
+			logger.debug("creating new LiveStreamPacketizerListener");
+			this.appInstance = appInstance;
+		}
+
+		public void onLiveStreamPacketizerCreate(ILiveStreamPacketizer liveStreamPacketizer, String streamName) {
+			logger.debug("onLiveStreamPacketizerCreate. stream: " + streamName);
+			if (!(liveStreamPacketizer instanceof LiveStreamPacketizerCupertino))
+				return;
+
+			LiveStreamPacketizerCupertino cupertinoPacketizer = (LiveStreamPacketizerCupertino) liveStreamPacketizer;
+			IMediaStream stream = this.appInstance.getStreams().getStream(streamName);
+
+			logger.info("Create [" + streamName + "]: " + liveStreamPacketizer.getClass().getSimpleName());
+			cupertinoPacketizer.setDataHandler(new LiveStreamPacketizerDataHandler(cupertinoPacketizer, streamName));
+		}
+
+		public void onLiveStreamPacketizerDestroy(ILiveStreamPacketizer liveStreamPacketizer) {
+			String streamName = liveStreamPacketizer.getAndSetStartStream(null).getName();
+			logger.debug("(" + streamName + ") onLiveStreamPacketizerDestroy");
+		}
+
+		public void onLiveStreamPacketizerInit(ILiveStreamPacketizer liveStreamPacketizer, String streamName) {
+			logger.debug("(" + streamName + ") onLiveStreamPacketizerInit");
+		}
+	}
+
+	public void onAppStart(IApplicationInstance applicationInstance) {
+		this.liveStreamPacketizerListener = new LiveStreamPacketizerListener(applicationInstance);
+		applicationInstance.addLiveStreamPacketizerListener(this.liveStreamPacketizerListener);
+		this.maxAllowedPTSDriftMillisec = applicationInstance.getProperties().getPropertyInt(MAX_ALLOWED_PTS_DRIFT_MILLISEC, DEFAULT_MAX_ALLOWED_PTS_DRIFT_MILLISEC);
+		logger.info("Modules.LiveStreamSettingsModule application instance started");
+	}
+
+	public void onStreamCreate(IMediaStream stream) {
+		PacketListener listener = new PacketListener();
+		WMSProperties props = stream.getProperties();
+		synchronized (props) {
+			props.setProperty(STREAM_ACTION_LISTENER_PROPERTY, listener);
+		}
+
+		stream.addLivePacketListener(listener);
+	}
+
+	public void onStreamDestroy(IMediaStream stream) {
+		PacketListener listener = null;
+		String streamName = stream.getName();
+		WMSProperties props = stream.getProperties();
+		synchronized (props) {
+			listener = (PacketListener) props.get(STREAM_ACTION_LISTENER_PROPERTY);
+		}
+		if (listener != null) {
+			stream.removeLivePacketListener(listener);
+			String entryId = Utils.getEntryIdFromStreamName(streamName);
+			this.removeEntryBaseSystemTime(entryId);
+			logger.info("remove PacketListener: " + stream.getSrc());
+		}
+
+	}
+
+	class PacketListener implements IMediaStreamLivePacketNotify {
+
+		private String entryId = null;
+
+		public PacketListener() {
+		}
+
+		public void onLivePacket(IMediaStream stream, AMFPacket thisPacket) {
+
+			if (stream.isTranscodeResult()) {
```

Comment:
better is to move the check for stream.isTranscodeResult() to the onStreamCreate rather to check that every single packet...